extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigDeploymentTemplate.java

```diff
@@ -47,9 +47,13 @@ public void validateConfigProperties(Map<String, Set<String>> properties) {
                 if (propertyClass.isArray() || propertyClass.getTypeParameters().length > 0) {
                     propertyClass = String.class;
                 }
-                if (!config.getOptionalValue(entry.getKey(), propertyClass).isPresent()) {
-                    throw new DeploymentException(
-                            "No config value of type " + entry.getValue() + " exists for: " + entry.getKey());
+                try {
+                    if (!config.getOptionalValue(entry.getKey(), propertyClass).isPresent()) {
+                        throw new DeploymentException(
+                                "No config value of type " + entry.getValue() + " exists for: " + entry.getKey());
+                    }
+                } catch (IllegalArgumentException e) {
+                    throw new DeploymentException(e);
```

Comment:
When can this new case happen?

Reply:
`config.getOptionalValue()` throws `IllegalArgumentException` if the property cannot be converted to the specified type. It was a bug in our integration - SM Config extension catches the exception as well: https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/inject/ConfigExtension.java#L93-L104